api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/TaskTranslatorImpl.java

```diff
@@ -173,52 +173,102 @@ private void setOpenmrsTaskFields(FhirTask openmrsTask, Task fhirTask) {
 		}
 		
 		if (!fhirTask.getInput().isEmpty()) {
-			openmrsTask.setInput(fhirTask.getInput().stream().map(this::translateToInputText).collect(Collectors.toSet()));
+			openmrsTask.setInput(fhirTask.getInput().stream().map(this::translateToInput).collect(Collectors.toSet()));
 		}
 		
 		if (!fhirTask.getOutput().isEmpty()) {
-			openmrsTask.setOutput(
-			    fhirTask.getOutput().stream().map(this::translateToOutputReference).collect(Collectors.toSet()));
+			openmrsTask.setOutput(fhirTask.getOutput().stream().map(this::translateToOutput).collect(Collectors.toSet()));
 		}
 		
 		openmrsTask.setName(FhirConstants.TASK + "/" + fhirTask.getId());
 		
 	}
 	
-	private FhirTaskOutput translateToOutputReference(Task.TaskOutputComponent taskOutputComponent) {
-		FhirReference outputReference = referenceTranslator.toOpenmrsType((Reference) taskOutputComponent.getValue());
-		Concept type = conceptTranslator.toOpenmrsType(taskOutputComponent.getType());
+	private FhirTaskOutput translateToOutput(Task.TaskOutputComponent taskOutputComponent) {
 		FhirTaskOutput output = new FhirTaskOutput();
-		
-		output.setValueReference(outputReference);
+		Concept type = conceptTranslator.toOpenmrsType(taskOutputComponent.getType());
 		output.setType(type);
 		
+		if (taskOutputComponent.getValue() instanceof Reference) {
```

Comment:
Are there any data types that might be in an input or output but we are not yet supporting? 

I think the possible types for input and output are quite large: https://build.fhir.org/datatypes.html#open, but obviously they don't all apply to the lab order workflows we are dealing with. I'm just wondering if we want to add any other support in this PR, or wait until there is a use case for it.

Reply:
Yeah , they are quite many .
we can always add more if need arises. i just wanted to support all [those supported ](https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/main/java/org/openmrs/module/fhir2/model/FhirTaskParam.java#L42-L52)by the TaskParam Class